packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
+Seller calls `encode` with an environment `{ n: ..., p: ... }`, indicating that the first inputs will spend `n` and the second inputs will spend `p`. Seller will use the expression `${Sale} Sale.exec(n, p)`. As the `exec` function of the `Sale` contract returns an array `[p, n]` the first output wil represent `p` and the second output will represent `n`. This is exactly the "crossover" transaction described above.
+
+```ts
+export class Sale extends Contract {
+  static exec(n: NFT, p: Payment) {
+    const [ownerN] = n._owners
+    const [ownerP] = p._owners
+    n.transfer(ownerP)
+    p.transfer(ownerN)
+    return [p, n]
+  }
+}
+```
 
-#### Partially Signed Transactions
+The first argument to the `exec` function is an nft of the following class.
 
-Todo.
+```ts
+export class NFT extends Contract {
+  constructor(name = '', symbol = '') {
+    super({ name, symbol })
+  }
 
-### Helper Classes
+  transfer(to: string) {
+    this._owners = [to]
+  }
+}
+```
 
-Buyer will use the smart contract below to create a smart object holding the satoshis to pay for the NFT.
+The second argument is created by Buyer to pay for the nft.
 
 ```ts
 class Payment extends Contract {
-  constructor(owner: string, _amount: number) {
-    super({ _owners: [owner], _amount })
+  constructor(_amount: number) {
+    super({ _amount })
   }
 
   transfer(to: string) {
     this._owners = [to]
   }
+
+  setAmount(a: number) {
+    this._amount = a
+  }
 }
 ```
 
-Seller will use this helper class to mock up the input that will later be used to spend Buyers's payment object.
+In the following sections, we explain the process of seller minting an NFT and creating a sales transaction as well as a buyer going through a purchase.
+
+### Minting an NFT
+
+To mint an nft, Seller can use the `computer.new` function.
 
 ```ts
-const randomPublicKey = '023e21361b53bb2e625cc1f41d18b35ae882e88d8d107df1c3711fa8bc54db8fed'
-const mockedRev = 'mock:0000000000000000000000000000000000000000000000000000000000000000:0'
+const nft = await seller.new(NFT, ['name', 'symbol'])
+```
+
+### Building the Sales Transaction
+
+The challenge with building the sales transaction is that the output that contains the payment is not yet known when the transaction is built. This is a problem because to evaluate an expression on the Bitcoin Computer either a smart object or the revision of an existing smart object is needed for all parameters of the call.
 
+For this situation the Bitcoin Computer provides a feature called mocking. A mock is a class that has the properties `_id`, `_rev`, `_root`, `_amount`, and `_owners`, and sets them to strings of the form `mock:<transaction-id>:<output-number>`. A mock does not have to extend from `Contract`. Each mocked object must have a distinct transaction id and output number.
```

Comment:
Instead of "For this situation", can we use "To address this situation" or more general "To handle such scenarios"?